### PR TITLE
[new release] phylogenetics (0.1.0)

### PR DESCRIPTION
**Pull request by pveber**
Algorithms and datastructures for phylogenetics

- Project page: <a href="https://github.com/biocaml/phylogenetics/">https://github.com/biocaml/phylogenetics/</a>

##### CHANGES:

- new Nucleotide_process module, implementing classical nucleotide
  evolution models
- Mutsel: added persistent positive selection
- Rate_matrix: added HKY85 model
- Alphabet, Alignment: a few more functions

### CODE REVIEW

File: packages/phylogenetics/phylogenetics.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  "core" {>= "v0.14.0"}
+  "dune" {>= "2.7.0"}
+  "gsl"
+  "lacaml" {>= "10.0.2"}
+  "menhir"
+  "ppx_deriving"
+  "printbox"
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/phylogenetics/phylogenetics.0.1.0/opam:22]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/phylogenetics/phylogenetics.0.1.0/opam [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  "alcotest" {with-test}
+  "angstrom-unix"
+  "biocaml" {>= "0.8.0"}
+  "core" {>= "v0.14.0"}
+  "dune" {>= "2.7.0"}
+  "gsl"
+  "lacaml" {>= "10.0.2"}
+  "menhir"
+  "ppx_deriving"
+  "printbox"
```
Comment (mseri) [inline packages/phylogenetics/phylogenetics.0.1.0/opam:19]:
```suggestion
  "printbox" {< "0.6"}
```

To fix 
```
# (cd _build/default/lib && /home/opam/.opam/4.08/bin/menhir newick_parser.mly --compile-errors newick_parser.messages) > _build/default/lib/newick_parser_errors.ml
# Read 3 sample input sentences and 2 error messages.
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/.phylogenetics.objs/byte -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/angstrom-unix -I /home/opam/.opam/4.08/lib/base -I /home/opam/.opam/4.08/lib/base/base_internalhash_types -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/base/md5 -I /home/opam/.opam/4.08/lib/base/shadow_stdlib -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/base_bigstring -I /home/opam/.opam/4.08/lib/base_quickcheck -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bin_prot -I /home/opam/.opam/4.08/lib/bin_prot/shape -I /home/opam/.opam/4.08/lib/biocaml/base -I /home/opam/.opam/4.08/lib/biocaml/ez -I /home/opam/.opam/4.08/lib/biocaml/unix -I /home/opam/.opam/4.08/lib/cfstream -I /home/opam/.opam/4.08/lib/core_kernel -I /home/opam/.opam/4.08/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.08/lib/core_kernel/binary_packing -I /home/opam/.opam/4.08/lib/fieldslib -I /home/opam/.opam/4.08/lib/gsl -I /home/opam/.opam/4.08/lib/jane-street-headers -I /home/opam/.opam/4.08/lib/lacaml -I /home/opam/.opam/4.08/lib/menhirLib -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_deriving/runtime -I /home/opam/.opam/4.08/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_expect/collector -I /home/opam/.opam/4.08/lib/ppx_expect/common -I /home/opam/.opam/4.08/lib/ppx_expect/config -I /home/opam/.opam/4.08/lib/ppx_expect/config_types -I /home/opam/.opam/4.08/lib/ppx_expect/evaluator -I /home/opam/.opam/4.08/lib/ppx_expect/matcher -I /home/opam/.opam/4.08/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_inline_test/config -I /home/opam/.opam/4.08/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/ppxlib/print_diff -I /home/opam/.opam/4.08/lib/printbox -I /home/opam/.opam/4.08/lib/re -I /home/opam/.opam/4.08/lib/re/perl -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/splittable_random -I /home/opam/.opam/4.08/lib/stdio -I /home/opam/.opam/4.08/lib/stringext -I /home/opam/.opam/4.08/lib/time_now -I /home/opam/.opam/4.08/lib/typerep -I /home/opam/.opam/4.08/lib/uri -I /home/opam/.opam/4.08/lib/variantslib -I /home/opam/.opam/4.08/lib/xmlm -I /home/opam/.opam/4.08/lib/zip -intf-suffix .ml -no-alias-deps -open Phylogenetics -o lib/.phylogenetics.objs/byte/phylogenetics__Tree.cmo -c -impl lib/tree.pp.ml)
# File "lib/tree.ml", line 205, characters 7-27:
# 205 |     |> PrintBox_text.output stdout
#              ^^^^^^^^^^^^^^^^^^^^
# Error: Unbound module PrintBox_text
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlopt.opt -w -40 -g -I lib/.phylogenetics.objs/byte -I lib/.phylogenetics.objs/native -I /home/opam/.opam/4.08/lib/angstrom -I /home/opam/.opam/4.08/lib/angstrom-unix -I /home/opam/.opam/4.08/lib/base -I /home/opam/.opam/4.08/lib/base/base_internalhash_types -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/base/md5 -I /home/opam/.opam/4.08/lib/base/shadow_stdlib -I /home/opam/.opam/4.08/lib/base64 -I /home/opam/.opam/4.08/lib/base_bigstring -I /home/opam/.opam/4.08/lib/base_quickcheck -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bigstringaf -I /home/opam/.opam/4.08/lib/bin_prot -I /home/opam/.opam/4.08/lib/bin_prot/shape -I /home/opam/.opam/4.08/lib/biocaml/base -I /home/opam/.opam/4.08/lib/biocaml/ez -I /home/opam/.opam/4.08/lib/biocaml/unix -I /home/opam/.opam/4.08/lib/cfstream -I /home/opam/.opam/4.08/lib/core_kernel -I /home/opam/.opam/4.08/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.08/lib/core_kernel/binary_packing -I /home/opam/.opam/4.08/lib/fieldslib -I /home/opam/.opam/4.08/lib/gsl -I /home/opam/.opam/4.08/lib/jane-street-headers -I /home/opam/.opam/4.08/lib/lacaml -I /home/opam/.opam/4.08/lib/menhirLib -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_deriving/runtime -I /home/opam/.opam/4.08/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_expect/collector -I /home/opam/.opam/4.08/lib/ppx_expect/common -I /home/opam/.opam/4.08/lib/ppx_expect/config -I /home/opam/.opam/4.08/lib/ppx_expect/config_types -I /home/opam/.opam/4.08/lib/ppx_expect/evaluator -I /home/opam/.opam/4.08/lib/ppx_expect/matcher -I /home/opam/.opam/4.08/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_inline_test/config -I /home/opam/.opam/4.08/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.08/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/ppxlib/print_diff -I /home/opam/.opam/4.08/lib/printbox -I /home/opam/.opam/4.08/lib/re -I /home/opam/.opam/4.08/lib/re/perl -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/splittable_random -I /home/opam/.opam/4.08/lib/stdio -I /home/opam/.opam/4.08/lib/stringext -I /home/opam/.opam/4.08/lib/time_now -I /home/opam/.opam/4.08/lib/typerep -I /home/opam/.opam/4.08/lib/uri -I /home/opam/.opam/4.08/lib/variantslib -I /home/opam/.opam/4.08/lib/xmlm -I /home/opam/.opam/4.08/lib/zip -intf-suffix .ml -no-alias-deps -open Phylogenetics -o lib/.phylogenetics.objs/native/phylogenetics__Tree.cmx -c -impl lib/tree.pp.ml)
# File "lib/tree.ml", line 205, characters 7-27:
# 205 |     |> PrintBox_text.output stdout
#              ^^^^^^^^^^^^^^^^^^^^
# Error: Unbound module PrintBox_text
```
From version 0.6 this is part of `printbox_text`

**Comment by pveber** [comment]
I have fixed the constraint on `printbox` upstream, as well as changed `dune` language version to 2.7 (to generate the correct `dune subst dev` line).

**Comment by mseri** [comment]
Thanks

**Comment by mseri** [comment]
The gsl compilation issue si not related to this PR

**Comment by pveber** [comment]
Thanks a lot Marcello!

**Pull request merged**